controllers/reconcilers/kserve_isvc_gateway_reconciler.go

```diff
@@ -0,0 +1,279 @@
+	if err := r.removeServerFromGateway(ctx, log, fmt.Sprintf("%s-%s", "https", isvc.Name)); err != nil {
+		log.V(1).Error(err, fmt.Sprintf("Failed to remove the Server(%s) from KServe local gateway in the %s namespace", fmt.Sprintf("%s-%s", "https", isvc.Name), meshNamespace))
+		errs = append(errs, err)
+	}
+
+	if len(errs) > 0 {
+		return fmt.Errorf("multiple errors: %v", errs)
+	}
+
+	return nil
+}
+
+func (r *KserveGatewayReconciler) Cleanup(ctx context.Context, log logr.Logger, isvcName string) error {
+	// NOOP - Resources should not be deleted until the kserve component is uninstalled.
+	return nil
+}
+
+func (r *KserveGatewayReconciler) processDelta(ctx context.Context, log logr.Logger, desiredGateway *istioclientv1beta1.Gateway, existingGateway *istioclientv1beta1.Gateway) (err error) {
+	comparator := comparators.GetGatewayComparator()
+	delta := r.deltaProcessor.ComputeDelta(comparator, desiredGateway, existingGateway)
+
+	if delta.IsUpdated() {
+		log.V(1).Info("Delta found", "update", desiredGateway.GetName())
+		gw := existingGateway.DeepCopy()
+		gw.Spec.Servers = append(existingGateway.Spec.Servers, desiredGateway.Spec.Servers[0])
+
+		if err = r.gatewayHandler.Update(ctx, gw); err != nil {
+			log.V(1).Error(err, fmt.Sprintf("Failed to add the Server(%s) from KServe local gateway in the istio-system namespace", desiredGateway.Spec.Servers[0].Port.Name))
+			return err
+		}
+		return nil
+	}
+	return nil
+}
+
+func (r *KserveGatewayReconciler) removeServerFromGateway(ctx context.Context, log logr.Logger, serverToRemove string) error {
+	gateway, err := r.gatewayHandler.Get(ctx, types.NamespacedName{Name: constants.KServeGatewayName, Namespace: meshNamespace})
+	if err != nil {
+		log.V(1).Error(err, "Failed to retrieve KServe local gateway in istio-system namespace")
+		return err
+	}
+
+	newServers := []*istiov1beta1.Server{}
+	for _, server := range gateway.Spec.Servers {
+		if server.Port.Name != serverToRemove {
+			newServers = append(newServers, server)
+		}
+	}
+
+	gateway.Spec.Servers = newServers
+	if err := r.gatewayHandler.Update(ctx, gateway); err != nil {
+		log.V(1).Error(err, "Failed to update KServe local gateway in istio-system namespace")
+		return err
+	}
+
+	return nil
+}
+
+func (r *KserveGatewayReconciler) copyServingCertSecretFromIsvcNamespace(ctx context.Context, sourceSecret *corev1.Secret, preDestSecret *corev1.Secret) error {
+	destinationSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-%s", sourceSecret.Name, sourceSecret.Namespace),
+			Namespace: meshNamespace,
+			Labels: map[string]string{
+				"opendatahub.io/managed": "true",
+			},
+		},
+		Data: sourceSecret.Data,
+		Type: sourceSecret.Type,
+	}
+
+	// Remove old secret if src secret is updated
+	if preDestSecret != nil {
+		if err := r.client.Delete(ctx, preDestSecret); err != nil {
+			return err
+		}
+	}
```

Comment:
@bartoszmajsak @rcernich This will delete a no longer useful TLS secret and create a new update one with the same name (e.g. at rotation).

Do you know if the ingress gateway will reload it without any changes to the `Gateway`? Or does it need to have a new name, and the `Gateway` must be updated to effectively reload it?